File: backend/app/models/appointment.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Boolean
from sqlalchemy.orm import relationship  # <- ADD THIS
from .database import Base

class Appointment(Base):
    __tablename__ = "appointments"
    
    id = Column(Integer, primary_key=True, index=True)
    patient_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    doctor_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    appointment_time = Column(DateTime, nullable=False)
    status = Column(String, default="scheduled")  # scheduled / completed / canceled
    appointment_type = Column(String, default="video")  # video / physical
    billing_amount = Column(Integer, default=30)
    payment_status = Column(String, default="unpaid")  # unpaid / paid
    notes = Column(String, nullable=True)
    is_active = Column(Boolean, default=True)

    # Relationships
    patient = relationship("User", foreign_keys=[patient_id])
    doctor = relationship("User", foreign_keys=[doctor_id])

File: frontend/smart_healthcare/lib/models/appointment.dart
class Appointment {
  final int id;
  final String patientEmail;
  final String doctorEmail;
  final DateTime appointmentTime;
  final String status;
  final String appointmentType;
  final int billingAmount;
  final String paymentStatus;
  final bool canJoinVideo;
  final String notes;

  Appointment({
    required this.id,
    required this.patientEmail,
    required this.doctorEmail,
    required this.appointmentTime,
    required this.status,
    required this.appointmentType,
    required this.billingAmount,
    required this.paymentStatus,
    required this.canJoinVideo,
    required this.notes,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] ?? 0,
      patientEmail: json['patient_email'] ?? '',
      doctorEmail: json['doctor_email'] ?? '',
      appointmentTime:
          DateTime.tryParse(json['appointment_time'] ?? '') ?? DateTime.now(),
      status: json['status'] ?? 'scheduled',
      appointmentType: json['appointment_type'] ?? 'video',
      billingAmount: json['billing_amount'] ?? 30,
      paymentStatus: json['payment_status'] ?? 'unpaid',
      canJoinVideo: json['can_join_video'] ?? false,
      notes: json['notes'] ?? '',
    );
  }

  bool get isVideo => appointmentType == 'video';
  bool get isScheduled => status == 'scheduled';
  bool get isPaid => paymentStatus == 'paid';
}
